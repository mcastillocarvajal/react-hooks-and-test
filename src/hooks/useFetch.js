import { useEffect, useRef, useState } from "react"


export const useFetch = ( url ) => {

    const isMounted = useRef(true)
    const [state, setState] = useState({ data: null, loading: true, err: null });

    useEffect(() => {
        return () => {
            isMounted.current = false
        }
    }, [])

    useEffect( () => {

        setState({ data: null, loading: true, err: null });

        fetch( url )
            .then( resp => resp.json() )
            .then( data => {

                if (isMounted.current) {
                    setState({
                        loading: false,
                        err: null,
                        data: data
                    })
                }

            })

    }, [ url ])

    return state;
}
