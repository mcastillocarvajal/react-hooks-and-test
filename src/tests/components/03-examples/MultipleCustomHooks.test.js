import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch'
import { useCounter } from '../../../hooks/useCounter'

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');


describe('Test for <MultipleCustomHooks />', () => {

    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    })
    
    test('should show component correctly', () => {

        useFetch.mockReturnValue({
            loading: true,
            data: null,
            err: null
        })

        const wrapper = shallow( <MultipleCustomHooks /> );
        expect(wrapper).toMatchSnapshot();
    })

    test('should show info', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Moises',
                quote: 'Hello world'
            }],
            loading: false,
            err: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb-0').text().trim() ).toBe('Hello world');
        expect( wrapper.find('footer').text().trim() ).toBe('Moises');
        // console.log(wrapper.html());
    })
    
    
})
