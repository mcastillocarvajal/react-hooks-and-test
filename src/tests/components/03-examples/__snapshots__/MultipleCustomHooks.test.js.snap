// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <MultipleCustomHooks /> should show component correctly 1`] = `
<div>
  <h1>
    Breaking Bad Quotes 
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    Loading...
  </div>
</div>
`;
